src/Raven.Server/Monitoring/ServerLimitsMonitor.cs

```diff
@@ -0,0 +1,214 @@
+        private static readonly string pid_max = "/proc/sys/kernel/pid_max";
+        private static readonly string max_map_count = "/proc/sys/vm/max_map_count";
+        private static readonly string threads_max = "/proc/sys/kernel/threads-max";
+
+        private readonly Logger _logger = LoggingSource.Instance.GetLogger<StorageSpaceMonitor>(nameof(ServerLimitsMonitor));
+        private readonly object _runLock = new();
+        private readonly NotificationCenter.NotificationCenter _notificationCenter;
+        private readonly List<string> _alerts = new List<string>();
+        private readonly Dictionary<string, long> _maxLimitsDictionary = new Dictionary<string, long>();
+
+        private Timer _timer;
+
+        public ServerLimitsMonitor(NotificationCenter.NotificationCenter notificationCenter)
+        {
+            _notificationCenter = notificationCenter;
+
+            if (PlatformDetails.RunningOnPosix == false || PlatformDetails.RunningOnMacOsx)
+            {
+                return;
+            }
+
+            _timer = new Timer(Run, null, CheckFrequency, CheckFrequency);
+        }
+
+        internal void Run(object state)
+        {
+            if (_notificationCenter.IsInitialized == false)
+                return;
+
+            if (Monitor.TryEnter(_runLock) == false)
+                return;
+
+            try
+            {
+                CheckMaxMapCountLimits();
+                CheckThreadsMaxLimits();
+                CheckPidMaxLimits();
+
+                if (_alerts.Count > 0)
+                {
+                    var warningMsg = string.Join(Environment.NewLine, _alerts);
+                    if (_logger.IsOperationsEnabled)
+                    {
+                        _logger.Operations($"Running close to OS limits detected:{Environment.NewLine}" + warningMsg);
+                    }
+
+                    var alert = AlertRaised.Create(
+                        null,
+                        "Running close to OS limits",
+                        warningMsg,
+                        AlertType.LowServerLimits,
+                        NotificationSeverity.Warning);
+
+                    _notificationCenter.Add(alert);
+                }
+            }
+            catch (Exception e)
+            {
+                if (_logger.IsOperationsEnabled)
+                    _logger.Operations($"Failed to run {nameof(ServerLimitsMonitor)}", e);
+            }
+            finally
+            {
+                _alerts.Clear();
+                Monitor.Exit(_runLock);
+            }
+        }
+
+        private void CheckPidMaxLimits()
+        {
+            var maxPidMax = GetMaxValueForLimit(pid_max);
+
+            using var proc = new Process();
+            proc.StartInfo.RedirectStandardOutput = true;
+            proc.StartInfo.RedirectStandardError = true;
+            proc.StartInfo.FileName = "ps";
+            proc.StartInfo.Arguments = @"-efL";
+            proc.Start();
+
```

Comment:
Let's not spawn a new process for this.
I'm not sure why we _care_ about `pid-max`, but that is not  good way to check this.

Reply:
removed